### DISCUSSION TITLE
Impossible layout with the APP dir ? NEXT.JS 13

### DISCUSSION THREAD

**Discussion by Florian-crg** [vercel/next.js#43429] [Help]
I have a website where the navbar can be sticked on top of the page or sticked on top of the page after a scroll. 

![Frame 165](https://user-images.githubusercontent.com/20863963/204144335-01cba7b0-2a6a-4c75-ac5c-7de835cdf59b.jpg)
![Frame 164](https://user-images.githubusercontent.com/20863963/204144330-caf227ef-7b0e-4b92-b858-fb29291ec4c5.jpg)

The banner is only on category pages. So only on these pages the navbar is sticked on top after scrolling.

For the moment my structure looks like that :
- app/
   - blog/
   - (shop)/
      - [category]/
- Navbar.tsx

Where should the banner stuff go ?
How can I have a Banner component that is on top of the global nav for category pages only ?

Edit: the banner component isn't a static image. The images on it depend on which category it is and it contains products data from the category.

**Comment by ihollarmide** [comment]
I don't totally understand you but how about you write a conditional using usePathname() to showcase the banner on top of the global nav for the selected category pages alone?

```
'use client'
import {usePathname} from 'next/navigation'

export default function Layout() {
  const routePathname = usePathname();
  
  return (
    <>
     {
       routePathname === "/category" ? <Banner /></GlobalNav />: <GlobalNav /><Banner />
     }
    </>
  )
}
```

**Reply by Florian-crg** [reply]
The banner depend on the data of the category pages. It contains dynamic data. It is not a static image. I think your solution would only work if it was a static image
